hector_mapping/src/HectorMappingRos.cpp

```diff
@@ -372,6 +381,30 @@ bool HectorMappingRos::mapCallback(nav_msgs::GetMap::Request  &req,
   return true;
 }
 
+bool HectorMappingRos::resetMapCallback(std_srvs::Trigger::Request  &req,
+                                        std_srvs::Trigger::Response &res)
+{
+  ROS_INFO("HectorSM Reset map service called");
+  slamProcessor->reset();
+  return true;
+}
+
+bool HectorMappingRos::pauseMapCallback(std_srvs::SetBool::Request  &req,
+                                        std_srvs::SetBool::Response &res)
+{
+  if (req.data && !pause_scan_processing_)
+  {
+    ROS_INFO("HectorSM Mapping paused");
+  }
+  else if (!req.data && pause_scan_processing_)
+  {
+    ROS_INFO("HectorSM Mapping no longer paused");
+  }
+  pause_scan_processing_ = req.data;
+  return true;
+}
```

Comment:
simple service callback that sets `pause_scan_processing_` to whatever was requested in `req`